addons/web_editor/static/lib/odoo-editor/src/commands/commands.js

```diff
@@ -525,8 +525,12 @@ export const editorCommands = {
                 } else {
                     font = [];
                 }
-            } else if (node.nodeType === Node.TEXT_NODE && isVisibleStr(node)) {
-                // Node is a visible text node: wrap it in a <font>.
+            } else if ((node.nodeType === Node.TEXT_NODE && isVisibleStr(node))
+                    || (node.nodeType === Node.ELEMENT_NODE &&
+                        ['inline', 'inline-block'].includes(getComputedStyle(node).display) &&
+                        isVisibleStr(node.textContent) &&
+                        !node.querySelector('font'))) {
+                // Node is a visible text or inline node without font: wrap it in a <font>.
```

Comment:
Mmmh... what about a block with a `a.btn-block` inside ? Wouldn't that give weird results since it is technially an inline but does not have display: inline ?

Reply:
It actually was a mess with all buttons: the button's style neutralized the applied color anyway. And indeed it does not make much sense to span a gradient color across a button.
I excluded `.btn` from this behavior so that they remain handled as they were before.

Reply:
@qsm-odoo LGTM. Unfortunately, special cases are inherent to the paradigm of the current editor.